grakn-graql/src/main/java/ai/grakn/graql/internal/hal/HALConceptRepresentationBuilder.java

```diff
@@ -221,4 +228,22 @@ private static void attachSingleGeneratedRelation(Representation currentHal, Con
             throw new RuntimeException("Unrecognized base type of " + type);
         }
     }
+
+    static void generateConceptState(Representation resource, Concept concept){
+
+        if (concept.isInstance()) {
+            Instance instance = concept.asInstance();
+            resource.withProperty(ID_PROPERTY, instance.getId().getValue())
+                    .withProperty(TYPE_PROPERTY, instance.type().getName().getValue())
+                    .withProperty(BASETYPE_PROPERTY, getBaseType(instance).name());
+        } else {
+            Type type = concept.asType();
+            resource.withProperty(ID_PROPERTY, type.getName().getValue())
```

Comment:
But in the case of a Type it's filled with a Type's Name. To limit confusion I propose that the ID should be the concept's ID regardless of it being a type or instance and a new property Name should be included in the resource for Types.